File: Step1/Driver.cpp
#include <iostream>
#include <fstream>

//#include "antlr4-runtime/antlr4-runtime.h"
#include "antlr4-runtime/LittleLexer.h"
#include "antlr4-runtime/LittleParser.h"

using namespace std;
using namespace antlr4;

int main(int argc, const char* argv[]){
    std::ifstream stream;
    stream.open("input.micro");

    ANTLRInputStream input(stream);
    LittleLexer lexer(&input);
    CommonTokenStream tokens(&lexer);
    
    ofstream outfile;
    outfile.open("output.micro");
    outfile << (&tokens);
    outfile.close;

    return 0;
}

File: Step3_python/Driver.py
##################################################################
## File: Driver.py                                              ##
## Date: 2/3/19                                                 ##
## Authors: Trevor Gahl, Skylar Tamke                           ##
## Purpose: To pass an input file to the generated antlr files. ##
##          Then take the output and print the token types and  ##
##          values.                                             ##
##################################################################

#############
## Imports ##
#############

import sys
from antlr4 import *
from LittleLexer import LittleLexer
from LittleListener import LittleListener
from LittleParser import LittleParser
from SymbolTableBuilder import SymbolTableBuilder


###################
## Main function ##
###################

def main(filename):
    # Create a lexer stream by passing filestream to generated lexer
    lexer = LittleLexer(FileStream(filename))
    # Create token stream from lexer
    stream = CommonTokenStream(lexer)
    # Initialize parser using token stream
    parser = LittleParser(stream)
    # Prevent parser error output (Tired of watching it fail)
    parser.removeErrorListeners()
    # Run parse tree using token stream initialized above - Step2+
    # Storing for use later - Step3
    tree = parser.program()
    # If program is valid print accepted, otherwise print not accepted


    #Step 2
    # if parser._syntaxErrors == 0:
    #     print("Accepted")
    # else:
    #    print("Not accepted")

    #Step 3
    symbol_table_builder = SymbolTableBuilder()
    walker = ParseTreeWalker()
    walker.walk(symbol_table_builder, tree)
    symbol_table_builder.print_symbol_table()

###############################################
## Python Main to read input and run program ##
###############################################

if __name__ == '__main__':
	filename = sys.argv[1]
	main(filename)
